langserver/references.go

```diff
@@ -96,9 +97,21 @@ func (h *LangHandler) handleTextDocumentReferences(ctx context.Context, conn JSO
 	} else {
 		users = h.importGraph.Search(defpkg)
 	}
+	findPackage := h.getFindPackageFunc()
 	lconf := loader.Config{
 		Fset:  fset,
 		Build: bctx,
+		FindPackage: func(bctx *build.Context, importPath, fromDir string, mode build.ImportMode) (*build.Package, error) {
+			// When importing a package, ignore any
+			// MultipleGoErrors. This occurs, e.g., when you have a
+			// main.go with "// +build ignore" that imports the
+			// non-main package in the same dir.
+			bpkg, err := findPackage(ctx, bctx, importPath, fromDir, mode)
```

Comment:
Yeah there is a bug lurking here. Given this findPackage is a caching implementation, when we start failing operations due to the context being cancelled, we don't want to cache those results.

This is a little tricky to fix. Two options:
* Document that we can fail findpackage FS operations and update our private implementation not to cache if the context is cancelled after the operation is done
* Not do this approach of failing bctx.

In general, BuildContext is used everwhere, and we cache failure in most places. So this approach actually introduces a bunch of bugs around caching.